### PR TITLE
feat(fraud): add "skip validation height" flag

### PR DESCRIPTION
**Pull request by srene**
# PR Standards

## Opening a pull request should be able to meet the following requirements

--

PR naming convention: https://hackmd.io/@nZpxHZ0CT7O5ngTp0TP9mg/HJP_jrm7A

---

Close #XXX

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/block.go [hunk 1]
```diff
@@ -14,10 +14,12 @@ import (
 // applyBlockWithFraudHandling calls applyBlock and validateBlockBeforeApply with fraud handling.
 func (m *Manager) applyBlockWithFraudHandling(block *types.Block, commit *types.Commit, blockMetaData types.BlockMetaData) error {
 	validateWithFraud := func() error {
-		if err := m.validateBlockBeforeApply(block, commit); err != nil {
-			m.blockCache.Delete(block.Header.Height)
-			// TODO: can we take an action here such as dropping the peer / reducing their reputation?
-			return fmt.Errorf("block not valid at height %d, dropping it: err:%w", block.Header.Height, err)
+		if m.Conf.SkipValidationHeight == block.Header.Height {
```
Comment (danwt) [inline block/block.go:17]:
```suggestion
		if m.Conf.SkipValidationHeight != block.Header.Height {
```

**Review by danwt — CHANGES_REQUESTED** [review]

File: proto/types/dymint/state.proto [hunk 2]
```diff
@@ -51,6 +51,7 @@ message State {
 
   int64 revision_start_height = 21;
 
+  uint32 prev_drs = 23;
```
Comment (omritoptix) [inline proto/types/dymint/state.proto:54]: what is this for ?

File: block/block.go (hunk 1 above)
Comment (srene) [reply block/block.go]: uups. fixed

File: proto/types/dymint/state.proto (hunk 2 above)
Comment (srene) [reply proto/types/dymint/state.proto]: nothing, it was from previous test. removed

File: types/state.go [hunk 3]
```diff
@@ -45,6 +45,8 @@ type State struct {
 
 	// LastHeaderHash is the hash of the last block header.
 	LastHeaderHash [32]byte
+
+	PrevDrs uint32
```
Comment (mtsitrin) [inline types/state.go:49]: remove here as well

Comment (srene) [reply types/state.go]: done

**Review by omritoptix — APPROVED** [review]

**Pull request merged**